Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/ProjectionIntegrityViolationDetector.php

```diff
@@ -178,35 +209,36 @@ public function tetheredNodesAreNamed(): Result
         return $result;
     }
 
-
-    /**
-     * @inheritDoc
-     */
     public function subtreeTagsAreInherited(): Result
     {
         $result = new Result();
 
         // NOTE:
         // This part determines if a parent hierarchy relation contains subtree tags that are not existing in the child relation.
         // This could probably be solved with JSON_ARRAY_INTERSECT(JSON_KEYS(ph.subtreetags), JSON_KEYS(h.subtreetags) but unfortunately that's only available with MariaDB 11.2+ according to https://mariadb.com/kb/en/json_array_intersect/
-        $hierarchyRelationsWithMissingSubtreeTags = $this->dbal->executeQuery(
-            'SELECT
-              ph.*
+        $hierarchyRelationsWithMissingSubtreeTagsStatement = <<<SQL
+            SELECT
+                ph.*
             FROM
-              ' . $this->tableNames->hierarchyRelation() . ' h
-              INNER JOIN ' . $this->tableNames->hierarchyRelation() . ' ph
-                ON ph.childnodeanchor = h.parentnodeanchor
-                AND ph.contentstreamid = h.contentstreamid
-                AND ph.dimensionspacepointhash = h.dimensionspacepointhash
+                {$this->tableNames->hierarchyRelation()} h
+                INNER JOIN {$this->tableNames->hierarchyRelation()} ph
+                    ON ph.childnodeanchor = h.parentnodeanchor
+                    AND ph.contentstreamid = h.contentstreamid
+                    AND ph.dimensionspacepointhash = h.dimensionspacepointhash
             WHERE
-              EXISTS (
-                SELECT t.tag FROM JSON_TABLE(JSON_KEYS(ph.subtreetags), \'$[*]\' COLUMNS(tag VARCHAR(30) PATH \'$\')) t WHERE NOT JSON_EXISTS(h.subtreetags, CONCAT(\'$.\', t.tag))
-              )'
-        )->fetchAllAssociative();
+                EXISTS (
+                    SELECT t.tag FROM JSON_TABLE(JSON_KEYS(ph.subtreetags), \'$[*]\' COLUMNS(tag VARCHAR(30) PATH \'$\')) t WHERE NOT JSON_EXISTS(h.subtreetags, CONCAT(\'$.\', t.tag))
```

Comment:
`\'` has to be unwrapped to `'` because were in heredoc now

Reply:
yes, that's wrong. Good catch – and I wonder why no test failed. I'll look into it

Reply:
For the record: The test _did_ fail (and some more) – they were just not executed in CI because this PR was based on a separate branch